packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1178,11 +1314,34 @@ void ArbCore::operator()() {
 
                 // Cache pre-sideload machines
                 if (last_assertion.sideload_block_number) {
+                    auto& output = core_machine->machine_state.output;
+
                     combined_machine_cache.timed_add(
                         std::make_unique<Machine>(*core_machine));
 
-                    if (core_machine->machine_state.output.arb_gas_used >=
-                        next_checkpoint_gas) {
+                    checkpoint_pruning_age_timestamp =
+                        seconds_since_epoch() -
+                        coreConfig.checkpoint_pruning_age_seconds;
+                    {
+                        std::lock_guard<std::mutex> lock(
+                            checkpoint_pruning_mutex);
+                        checkpoint_pruning_gas_used =
+                            unsafe_checkpoint_pruning_gas_used;
+                    }
```

Comment:
Maybe better to just set these where they're used and set a generic `perform_pruning` here?

Reply:
:+1: